### ISSUE TITLE
SO3FunRBF.approximate(nodes,values,...) errors where interpolate accepts the same arguments

### ISSUE THREAD

**Issue by ralfHielscher** [mtex-toolbox/mtex#2595]
`SO3FunRBF.interpolate(nodes,values,...)` works, but the same arguments given to
`SO3FunRBF.approximate` error out before any solver runs.

## Reproducer

```matlab
fname = fullfile(mtexDataPath,'orientation','dubna.csv');
[ori,S] = orientation.load(fname,'columnNames',{'phi1','Phi','phi2','values'});
psi = SO3DeLaValleePoussinKernel('halfwidth',5*degree);

SO3FunRBF.interpolate(ori,S.values,'kernel',psi)          % ok
SO3FunRBF.approximate(ori,S.values,'kernel',psi,'odf')    % errors
```

```
Dot indexing is not supported for variables of this type.
  in orientation/cat        (cat.m:18)
  in quaternion/horzcat     (horzcat.m:4)
  in SO3FunRBF.approximate  (approximate.m:89)
```

## Cause

`approximate(f,varargin)` assumes its first argument is an `SO3Fun` or a
`function_handle`. Given an `orientation`, the values land in `varargin` and `f`
stays an `orientation`, so line 89

```matlab
varargin = [varargin,'mean',f.mean];
```

takes `mean` of an *orientation* — a mean orientation, not a number — and the
concatenation then fails.

Either `approximate` should accept the `(nodes,values)` form and hand it to
`interpolate` (which already handles it), or it should reject a non-`SO3Fun`
first argument with a message that says so. The docstring's Syntax block lists
only the `approximate(f,...)` forms, so the second reading is defensible — but
`tests/check_SO3FunRBFApproximation.m` calls the `(nodes,values)` form in two
places, so something expected it to work.

## How this surfaced

It was masked: `check_SO3FunRBFApproximation` aborted earlier at its harmonic
scale-equivariance block (#2588). With that fixed the test runs on and reaches
this at line 98. Verified identical with and without the #2588 fix, so it is
independent and pre-existing.

The test therefore stays parked at `tests/` root instead of moving into `core/`.

**Comment by ralfHielscher** [comment]
Fixed on `develop` in 7563d692e.

`approximate` now hands the `(nodes,values)` form straight to
`SO3FunRBF.interpolate`. That is what `approximate`'s own description says the
spatial method does anyway — *"Choose a grid of orientations R_i and evaluate f. Do
SO3FunRBF.interpolation afterwards"* — `interpolate` takes the same option names,
and the result is identical to calling it directly (weights agree to 0). The
`SO3Fun` and `function_handle` routes are untouched.

I picked delegation over the other option in the report — rejecting a non-`SO3Fun`
first argument with a clear error — because `check_SO3FunRBFApproximation` calls
this form twice with accuracy assertions on the result, so something clearly
intended it to work. Easy to switch to the strict reading if you would rather keep
`approximate` to one signature.

```
approximate(ori,values,'odf')       err 0.00061   (assertion wants < 0.02)
approximate(ori,noisy,'odf')        err 0.04961   (assertion wants < 0.06)
same as interpolate                 max|weight difference| 0
```

`check_SO3FunRBFApproximation` passes end to end now and leaves the `tests/` root
parking area — into `slow/` rather than `core/`, since it takes about 20 s, a third
of the whole core budget.

**One thing worth knowing, found on the way.** Its last assertion could not be moved
into a tier as written. The noisy Dubna case draws *unseeded* noise and sits exactly
on its own threshold — across twelve seeds the error runs 0.0494 to 0.0504 against a
bound of 0.05, so it would have failed on **five of twelve** runs. That was invisible
while the file was parked and never executed. It is seeded now, with the bound at
0.06: the case is about 5% noise not blowing the fit up, and that holds whatever the
draw.
